lms/djangoapps/grades/tests/test_subsection_grade_factory.py

```diff
@@ -134,13 +141,15 @@ def test_update_with_override(self):
         PersistentSubsectionGradeOverride.update_or_create_override(
             UserFactory(),
             persistent_grade,
-            earned_graded_override=0,
+            earned_graded_override=graded_earned_override,
             earned_all_override=0,
+            possible_graded_override=graded_possible_override,
             feature=PersistentSubsectionGradeOverrideHistory.GRADEBOOK,
         )
 
         # Now, even if the problem scores interface gives us a 2/3,
         # the subsection grade returned should be 0/3 due to the override.
+        earned_value = graded_earned_override or persistent_grade.earned_graded
```

Comment:
I think it would be better if you stored the return value of `update_or_create_override()` inside an `override` variable and then did the assert below as `self.assert_grade(grade, override.earned_graded_override, 3)`.  The way you're doing it here exposes implementation details inside the test code.

Reply:
I tried this and found I cannot really do what you suggested. This is because the return value of `update_or_created_override()` will only return the override object. But the value from that object is only 1 piece of the puzzle. I need to handle the null case from override. That's the point of my code. Please code review my newest change on this test case and see if you agree with it.